src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -66611,9 +66563,15 @@ class C
                 // (15,12): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
                 //     static U?[] F1<T, U>() where U : T => throw null!; // error
                 Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "U?").WithLocation(15, 12),
+                // (17,12): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
+                //     static U?[] F3<T, U>() where T : struct where U : T => throw null!;
+                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "U?").WithLocation(17, 12),
```

Comment:
>Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "U?").WithLocation(17, 12), [](start = 16, length = 88)

nit: this is a cascading diagnostic. The root cause is below ("Type parameter 'T' has the 'struct' constraint so 'T' cannot be used as a constraint for 'U'"). Might there be a simple way to suppress?

Reply:
> nit: this is a cascading diagnostic. The root cause is below ("Type parameter 'T' has the 'struct' constraint so 'T' cannot be used as a constraint for 'U'"). Might there be a simple way to suppress?

Unfortunately there is no easy way to do that. We are not tracking what constraints were removed as invalid and what impact they otherwise would have if they were valid. Also, now we have a consistent experience across all usages of ```U?```. For example, in the code below prior to this change the CS8627 would be reported only inside method body, but not inside method signature. With this change, method signature gets the same treatment. 
```
    class C1
    {
#nullable enable
        static U?[] F3<T, U>() where T : struct where U : T
        {
            U? x = default;
            Console.WriteLine(x);
            throw null!;
        }
    }
```

---
In reply to: [280618191](https://github.com/dotnet/roslyn/pull/35375#discussion_r280618191) [](ancestors = 280618191)